### DISCUSSION TITLE
GetServerSideProps fetching image elements running twice

### DISCUSSION THREAD

**Discussion by vijaystroup** [vercel/next.js#19545] [Help]
# Bug report

## Describe the bug

When using GetServerSideProps to fetch and dangerouslySetInnerHTML from markdown file with img tag, the GetServerSideProps function runs twice, causing it to call my api again making an unnecessary request with bad data as a query.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

```ts
export default function Article({ article, text }) {
  return (
    <div className='content'>
      <h1 className='title'>{article.title}</h1>
      <h2 className={style.author}>{article.author} • {article.date}</h2>
      <div className={style.text} dangerouslySetInnerHTML={{ __html: text }}></div>
    </div>
  )
}

async function markdownToHtml(text) {
  const result = await remark().use(lint).use(html).process(text)
  return result.toString()
}

export const getServerSideProps: GetServerSideProps = async ctx => {
  let articleTitle = (ctx.params.article as string).replace(/(-)/g, ' ')
  articleTitle = articleTitle.replace(/(^\w{1})|(\s+\w{1})/g, l => l.toUpperCase())

  const res = await fetch(`http://localhost:8000/article?article_id=${articleTitle}`)
  const doc = await res.json()
  const article = doc.article
  const text = await markdownToHtml(doc.text)

  return {
    props: {
      article,
      text
    }
  }
}
```

## Expected behavior

I expect for the GetServerSideProps to run twice, however it runs the first time with a correct call to the api with no errors, but then runs again with the `articleTitle` variable now holding the alt text of the image in the markdown file.

## Screenshots

The markdown file:
![image](https://user-images.githubusercontent.com/38572582/100292923-fbe5df00-2f4f-11eb-965f-abd0c1b75654.png)

## System information

- OS: Ubuntu 20.04 WSL
- Browser (if applies) Chrome
- Version of Next.js: 9.5.5
- Version of Node.js: v14.13.1
- Deployment: next start

## Additional context

When the markdown file does not contain an img element, it does not call GetServerSideProps twice and behaves as expected calling the api only one time and receiving all the correct data.

**Comment by djleonskennedy** [comment]
Could be related to https://github.com/vercel/next.js/discussions/13064 still have this issue

**Comment by mimiqkz** [comment]
console.count() and I confirm it runs twice. 
#13064 seems to be unanswered,
The problem still persist today
